plotly/tests/test_core/test_graph_objs/test_scatter.py

```diff
@@ -26,8 +25,8 @@ def test_validate():
     Scatter().validate()
 
 
-@raises(PlotlyError)
-def test_validate_error():
-    scatter = Scatter()
-    scatter['invalid'] = 'something'
-    scatter.validate()
+# @raises(PlotlyError)
+# def test_validate_error():
+#     scatter = Scatter()
+#     scatter['invalid'] = 'something'
+#     scatter.validate()
```

Comment:
🚨 Breaking Change (immediate validation)
Let's keep this test but remove the `scatter.validate()` as the new expected behaviour is for `scatter['invalid'] = 'something'` to raise an error.

similarly, let's add a new test that uses the dot syntax to add the property:
```
scatter.invalid = 'something'
```

Reply:
and we'll want a whole set of new validation tests since https://github.com/plotly/plotly.py/pull/942 is much more strict. So, for each of the possible value types (`enumerated, boolean, number, integer, string, color, colorlist, colorscale, angle, subplotid, flaglist, any, info_array`), we'll want to attempt adding a type that is _not_ the correct one and verify that an exception is thrown. We'll also want to verify that all possible _correct_ values are possible. 

For example, here are examples of exceptions that should be thrown:
```
go.Scatter(x=1)
go.scatter.Marker(color=3)
go.scatter.Marker(color='redz')
go.scatter.Marker(size='a')
```
and here are examples of valid ones we should check:

```
valid_colors = [
    # Name
    'hotpink',
    'transparent',

    # Hexadecimal syntax',
    #f09',
    #F09',
    #ff0099',
    #FF0099',

    # Functional syntax',
    'rgb(255,0,153)',
    'rgb(255, 0, 153)',
    'rgb(100%,0%,60%)',
    'rgb(100%, 0%, 60%)',
    'rgb(255 0 153)',

    # Hexadecimal syntax with alpha value',
    #f09f',
    #F09F',
    #ff0099ff',
    #FF0099FF',

    # Functional syntax with alpha value',
    'rgb(255, 0, 153, 1)',
    'rgb(255, 0, 153, 100%)',

    # Whitespace syntax',
    'rgb(255 0 153 / 1)',
    'rgb(255 0 153 / 100%)',

    # Functional syntax with floats value',
    'rgb(255, 0, 153.6, 1)',
    'rgb(1e2, .5e1, .5e0, +.25e2%)',

    # Hexadecimal syntax
    '#3a30',                    #   0% opaque green 
    '#3A3F',                    # full opaque green 
    '#33aa3300',                #   0% opaque green 
    '#33AA3388',                #  50% opaque green 

    # Functional syntax
    'rgba(51, 170, 51, .7)',    #  70% opaque green 
    'rgba(51, 170, 51,  1)',    # full opaque green 

    # Whitespace syntax
    'rgba(51 170 51 / 0.4)',    #  40% opaque green 
    'rgba(51 170 51 / 40%)',    #  40% opaque green 

    # Functional syntax with floats value
    'rgba(255, 0, 153.6, 1)',
    'rgba(1e2, .5e1, .5e0, +.25e2%)',

    # HSL
    'hsl(30, 10, 10)',
]

for color in valid_colors:
    go.scatter.Marker(color=color)
    go.scatter.Marker(color=[color]*5)

go.scatter.Marker(color=1)
go.scatter.Marker(color=[1]*5)
```

Reply:
I actually wrote a pretty extensive set of tests for the new validators. I hadn't brought them over yet because I wrote them using pytest.  I just added them in commit 45ec0846bf7bdb385135cda600d2367701eaeb20 so you can take a look.

Run them with
```
~> pip install pytest
~> pytest _plotly_utils/tests/
```

So, we can either port them out of pytest, or consider adopting pytest.

Reply:
Ah fantastic! I'm OK with adopting pytest, even if that means that some of our tests are in `pytest` and others are in nosetest. This is where the tests are invoked on CI, so let's add the `pytest` commands to this file here: https://github.com/plotly/plotly.py/blob/master/tox.ini#L74